examples/citrix_adc/README.md

```diff
@@ -0,0 +1,369 @@
+
+<!-- **DW:** The original paragraph above wasn't clear to me; in my attempt to undersand it, I've written a new para. If I've lost the technical meaning, it's because I couldn't follow the original logic. Some of the questions I had from the original were these: Which parts of the explanation are Terraform's and which parts are Venafi...because the first half of the original sentence made it sound like Terraform has an automated process already for generating and installing certs, and so why woud you need Venafi? But I knew that's not true. So I wondered if it was saying that the Venafi Provider, as a service component of Terraform, is creating/installing the certs? In short, I wasn't clear which parts are us and which parts are Terraform, etc. And understanding that will I think help users stay oriented to "who's doing what" as they prepare to test drive your example. -->
+
+<!-- NOTE: DW moved the content below from what was called the Scenario Introduction section, which was below the Getting Started section, per our discussion on 31 March; we'd agreed that it makes more sense here and that it interupted the flow of the document in it's old location. --> 
+
+Later in this example, you'll generate a certificate for ``demo-citrix.venafi.example`` using the _Venafi Provider for Hashicorp Terraform_ with either Venafi Trust Protection Platform (TPP) or Venafi Cloud. Then after adding them to your Citrix ADC resources, you'll use them in the ADC node. And finally, you'll configure the service group members and [bind them](https://docs.citrix.com/en-us/citrix-adc/current-release/load-balancing/load-balancing-manage-large-scale-deployment/configure-service-groups.html#bind-a-service-group-to-a-virtual-server) to your ADC node.
+
+> **NOTE** While we'll be using a ``Round robin`` balancing method in our ADC configuration, keep in mind that there are other [methods](https://docs.citrix.com/en-us/citrix-adc/current-release/load-balancing/load-balancing-customizing/assign-weight-services.html) that might be more suitable for your specific use case.
+
+![scenario](scenario.png "Scenario")
+
+### About retrieving a certificate using the _Venafi Provider for Terraform_
+
+> **NOTE** The only purpose of the credentials used in this example is illustrative, in a real life scenario they must be considered as **weak** and **insecure**.
+<!-- This seems like a strange place for this note; is this about the generic creds used in the steps below? And is the intent to tell users that they shouldn't use simple passwords (e.g. "password") in production environments? Once I understand the purpose, I can suggest some changes... -->
```

Comment:
No, this has to do with the development environment for this example. In a production enviroment, the credentials would signed by a trusted CA, but in this case the credentials are self-signed by the platform you are requesting the credentials:

![venafi-cloud-example](https://user-images.githubusercontent.com/77068025/113330899-7cd92c80-92dc-11eb-8db8-e13b28bce0a2.png)

And self-signed credentials are considered dangerous and insecure:

https://www.globalsign.com/en/ssl-information-center/dangers-self-signed-certificates

Reply:
Great point. So borrowing from our own documentation on self-signed certs, I suggest you add a note like this:

>**Best Practice** In general, be careful when using self-signed certificates because of the inherent risks of no identity verification or trust control. The public and private keys are both held by the same entity. Also, self-signed certificates cannot be revoked; they can only be replaced. If an attacker has already gained access to a system, the attacker can spoof the identity of the subject. Of course, CAs can revoke a certificate only when they discover the compromise.